core/Enclave/openssl_operation.cpp

```diff
@@ -58,6 +58,16 @@
 #define SM4_NO_PAD 0
 #define SM4_PAD 1
 
+static void pkcs7_unpadding(ehsm_data_t *plaintext, int length)
+{
+    int pad = plaintext->data[length - 1];
```

Comment:
If the plaintext is ended with a newline character (`\n`, `0xA` in ASCII) and its length is the multiple of 16, this function will remove the last ten characters of the ciphertext.